src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -47,7 +47,8 @@ public class IndexUtil {
     private static final Version MINIMAL_SUPPORTED_VERSION_FOR_MODEL_METHOD_COMPONENT_CONTEXT = Version.V_2_13_0;
     private static final Version MINIMAL_SUPPORTED_VERSION_FOR_RADIAL_SEARCH = Version.V_2_14_0;
     private static final Version MINIMAL_SUPPORTED_VERSION_FOR_METHOD_PARAMETERS = Version.V_3_0_0;
-    private static final Map<String, Version> minimalRequiredVersionMap = initializeMinimalRequiredVersionMap();
+    // public so neural search can access it
+    public static final Map<String, Version> minimalRequiredVersionMap = initializeMinimalRequiredVersionMap();
```

Comment:
Out of curiosity, could neural search not access the map through the public methods below?

Reply:
So the methods below use KNNClusterState. And for neural search it uses NeuralClusterState. so can't reuse those methods. Since it was static I went with making it public instead of a getter

Reply:
Makes sense, thanks for the insight